packages/seed/src/SnippetTestBuilder.ts

```diff
@@ -0,0 +1,90 @@
+import path from "path";
+import tmp from "tmp-promise";
+
+
+export class SnippetTestBuilder {
+    public language: GenerationLanguage;
+
+    constructor(language: GenerationLanguage) {
+        this.language = language;
+    }
+
+    async buildSnippetTests(snippets: FernGeneratorExec.Snippets): Promise<tmp.DirectoryResult[]> {
+        const directories: tmp.DirectoryResult[] = [];
+        for (const endpointSnippet of snippets.endpoints) {
+            if (endpointSnippet.snippet.type !== this.language) {
+                for (const dir of directories) {
+                    await dir.cleanup();
+                }
+                throw new Error(`Failed to build snippet tests; expected language ${this.language}, but found snippet with language ${endpointSnippet.snippet.type}`);
+            }
+            const dir = await tmp.dir({unsafeCleanup: true});
+            switch (endpointSnippet.snippet.type) {
+                case "python": await this.buildSnippetForPython(dir, endpointSnippet.snippet);
+            }
+            directories.push(dir);
+        }
+        return directories;
+    }
+
+    async buildSnippetForPython(
+        dir: tmp.DirectoryResult,
+        snippet: FernGeneratorExec.PythonEndpointSnippet,
+    ): Promise<void> {
+        // This is prone to drift - if we change any of these values in fern-python, we'll
+        // need to update these values, too.
+        //
+        // However, copying the pyproject.toml from the generated output also isn't ideal
+        // because it requires find/replacing values that must be unique (i.e. we would
+        // need to update the 'name' and remove the 'packages' field in the tools.poetry
+        // section for this to work, as well as add the 'seed' dependency override in
+        // the tool.poetry.dependencies section.
+        //
+        // Although less than ideal, including a separate copy of the pyproject.toml
+        // that only specifies the values we need is simplest.
+        await fs.writeFile(
+            path.join(dir.path, "pyproject.toml"),
+            [
+                "[tool.poetry]",
+                "name = \"snippet\"",
+                "version = \"0.0.0\"",
+                "description = \"\"",
+                "authors = []",
+                "",
+                "[tool.poetry.dependencies]",
+                "python = \"^3.7\"",
+                "seed = { path = \"/generated\" }",
+                "",
+                "[tool.poetry.dev-dependencies]",
+                "mypy = \"0.971\"",
+                "pytest = \"^7.4.0\"",
+            ].join("\n"),
+        );
+        await fs.mkdir(path.join(dir.path, "src"));
+        await fs.writeFile(
+            path.join(dir.path, "src", "sync_client.py"),
+            snippet.syncClient,
+        );
+        await fs.writeFile(
+            path.join(dir.path, "src", "async_client.py"),
+            formatAsyncPythonSnippet(snippet.asyncClient),
+        );
+    }
+}
```

Comment:
can we have the generator author specify this in `seed.yml` ? They can also just point to a file so that it doesn't clutter up `seed.yml` itself

Reply:
Good call - if we continue to go down this route then that's a good call. @armandobelardo suggested moving even more of this into the Python generator so I'll continue to explore that in the meantime.